### PR TITLE
Redo Test Examples for 1.4

### PR DESCRIPTION
**Pull request by matjin**
This PR adds a thread safety violation example, redoes the Resource Leak tests for the implementation of Pulse Resource Leaks, and also adds a Taint example identifying SQL injections using Pulse. It also updates all the projects to .NET 6.0.

### CODE REVIEW

**Review by xinshiMSFT — APPROVED** [review]

**Comment by xi-liu-ds** [comment]
Overall it looks good. My general suggestion is we can add a README to explain all the errors that supposed to found in these examples.

**Comment by matjin** [comment]
> Overall it looks good. My general suggestion is we can add a README to explain all the errors that supposed to found in these examples.

Are you thinking maybe better comments? I feel like a separate README might create distraction.

**Comment by xi-liu-ds** [comment]
> > Overall it looks good. My general suggestion is we can add a README to explain all the errors that supposed to found in these examples.
> 
> Are you thinking maybe better comments? I feel like a separate README might create distraction.

Yeah, that is good idea to comment explanations inline as docstrings.

File: Cilsil.Test/Cilsil.Test.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (xi-liu-ds) [inline Cilsil.Test/Cilsil.Test.csproj:4]: For consistency, we should update [TestRunManager.cs](https://github.com/microsoft/infersharp/blob/64dc14a7542acfb042183482a22cabe4e66eeff8/Cilsil.Test/TestRunManager.cs#L31), [CONTRIBUTING](https://github.com/microsoft/infersharp/blob/98371c63612a22c35cc9148ebab3b149d989eab6/CONTRIBUTING.md#prerequisites), and [TestProject.csproj](https://github.com/microsoft/infersharp/blob/98371c63612a22c35cc9148ebab3b149d989eab6/Cilsil.Test/Assets/TestProject.csproj) to adopt net6.0

**Comment by matjin** [comment]
I've added comments documenting everything.

**Review by xi-liu-ds — APPROVED** [review]

**Pull request merged**